openapi/code/service.go

```diff
@@ -206,6 +206,17 @@ func (svc *Service) newRequest(params []openapi.Parameter, op *openapi.Operation
 		request.Description = op.Summary
 		svc.Package.define(request)
 	}
+	isOnlyUsedAsRequest := true
+	for _, e := range svc.Package.types {
+		for _, f := range e.fields {
+			if f.Entity == request {
+				// this type is not only a request placeholder, but is also
+				// part of other type definitions.
+				isOnlyUsedAsRequest = false
+			}
+		}
+	}
+	request.IsRequestOnly = isOnlyUsedAsRequest
```

Comment:
Doesn't this need to be done as a postprocessing step? The order in which requests are defined is arbitrary AFAIK, so this could accidentally set types as request only if their uses are not yet processed

Reply:
> Doesn't this need to be done as a postprocessing step?

it could be in `NonPlaceholderTypes()` method. may add some false positives, but will try that.